datalad/support/repo.py

```diff
@@ -249,3 +249,32 @@ class RepoInterface(object):
 
     # Test!
     pass
+
+
+def path_based_str_repo(cls):
+    """A helper decorator for a class to define str and repo based on its .path
+
+    For the rationale/discussion on why to bother distinguishing the two is
+    in https://github.com/datalad/datalad/pull/4439 . The idea  is that
+    `__str__` should provide cut/pasteable to shell representation of the path,
+    with all necessary escapes for characters shell might care about.
+    `__repr__` lives to its promise to provide string representation consumable
```

Comment:
Hrm, probably not worth changing, but I think "promise" is a bit strong, based on my understanding of `repr` conventions and anything I see in the [official docs](https://docs.python.org/3/reference/datamodel.html?highlight=iadd#object.__repr__).

Reply:
yeah, could have just removed "lives to its promise" I guess. would do if further changes to happen.

Reply:
oh spotted that I haven't addressed comment above, so will push `no ci` commit with those now